src/common/utility.ts

```diff
@@ -627,6 +627,61 @@ export class Utility {
             validateFunc, Constants.moduleNameDft);
     }
 
+    public static async checkDockerState(outputChannel: vscode.OutputChannel) {
+        let state: DockerState;
+        let errorMsg: any;
+        try {
+            await Executor.executeCMD(outputChannel, "docker", { shell: true }, "version");
+            state = DockerState.Runing;
+        } catch (error) {
+            errorMsg = error;
+            if (os.platform() === "win32") {
+                if (error.message.indexOf("is not recognized as an internal or external command") > -1) {
+                    state = DockerState.NotInstalled;
+                } else if (error.message.indexOf("This error may also indicate that the docker daemon is not running") > -1) {
+                    state = DockerState.NotRunning;
+                } else {
+                    state = DockerState.Unknown;
+                }
+            } else if (os.platform() === "linux" || os.platform() === "darwin") {
+                if (error.message.indexOf("command not found") > -1 || error.message.match(/Command '.*?' not found/)) {
+                    state = DockerState.NotInstalled;
+                } else if (error.message.indexOf("Is the docker daemon running") > -1) {
+                    state = DockerState.NotRunning;
+                } else if ("permission denied") {
+                    state = DockerState.PermissionDenied;
+                } else {
+                    state = DockerState.Unknown;
+                }
+            }
+        }
+
+        if (errorMsg) {
+            const install: vscode.MessageItem = { title: "Install Docker" };
+            const troubleshooting: vscode.MessageItem = { title: "Troubleshooting" };
+            const cancel: vscode.MessageItem = { title: "Cancel" };
+            let input: vscode.MessageItem;
+            let helpUrl: string;
+            if (state === DockerState.NotInstalled) {

```

Comment:
there is case state is not set. Shall we handle it, or state should assigned a default value?